### PR TITLE
Remove NaN from SEMAlg

### LINKED ISSUE #1902: A Bug in SEMEstimator.py
The masks are being used in different ways in SEM.py and SEMEstimator.py

in [SEM.py](https://github.com/pgmpy/pgmpy/blob/f4a0485668f437182d08abdf4b33d5be1b7c42d1/pgmpy/models/SEM.py#L882C1-L886C10)
```
# Masks represent the parameters which need to be learnt while training.
self.B_mask = np.multiply(np.where(self.B_fixed_mask != 0, 0.0, 1.0), self.B)
self.zeta_mask = np.multiply(
    np.where(self.zeta_fixed_mask != 0, 0.0, 1.0), self.zeta
)
```

in [SEMEstimator.py](https://github.com/pgmpy/pgmpy/blob/f4a0485668f437182d08abdf4b33d5be1b7c42d1/pgmpy/estimators/SEMEstimator.py#L75)
```
B_masked = torch.mul(B, self.B_mask) + self.B_fixed_mask
B_inv = pinverse(self.B_eye - B_masked)
zeta_masked = torch.mul(zeta, self.zeta_mask) + self.zeta_fixed_mask
```
[and here](https://github.com/pgmpy/pgmpy/blob/f4a0485668f437182d08abdf4b33d5be1b7c42d1/pgmpy/estimators/SEMEstimator.py#L325C1-L327C1)
```
B = params["B"] * self.B_mask + self.B_fixed_mask
zeta = params["zeta"] * self.zeta_mask + self.zeta_fixed_mask
```

Using the test cases in `TestSEMEstimator`, 
`self.B_fixed_mask` have NaN's. Leading to failure in pinverse().

Do we expect the masking to act like SEM:
```
B = torch.mul(torch.where(self.B_fixed_mask != 0, 0.0, 1.0), params["B"])
zeta = torch.mul(torch.where(self.zeta_fixed_mask != 0, 0.0, 1.0), params["zeta"])
```
?

But still I could not make the [test cases](https://github.com/pgmpy/pgmpy/blob/f4a0485668f437182d08abdf4b33d5be1b7c42d1/pgmpy/tests/test_estimators/test_SEMEstimator.py#L101-L151) work, I was getting [negative determinant](https://github.com/pgmpy/pgmpy/blob/f4a0485668f437182d08abdf4b33d5be1b7c42d1/pgmpy/estimators/SEMEstimator.py#L343C20-L343C44) for `sigma_hat`.

### PR DESCRIPTION
**Pull request by jihyeseo**
### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [ ] Make sure you are making a pull request against the **dev branch** (left side). Also you should start *your branch* off *our dev*.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.

### Issue number(s) that this pull request fixes
- Fixes #1902 

### List of changes to the codebase in this pull request
- 
-
-

### CODE REVIEW

File: pgmpy/tests/test_estimators/test_SEMEstimator.py [hunk 1]
```diff
@@ -103,14 +103,14 @@ def test_demo_estimator_random_init(self):
         estimator = SEMEstimator(self.demo)
```
Comment (jihyeseo) [inline pgmpy/tests/test_estimators/test_SEMEstimator.py]: This test still fails. So it remains to be skipped.

**Comment by ankurankan** [comment]
@jihyeseo Wouldn't it be more suitable to make the changes at: https://github.com/pgmpy/pgmpy/blob/f4a0485668f437182d08abdf4b33d5be1b7c42d1/pgmpy/models/SEM.py#L592 and https://github.com/pgmpy/pgmpy/blob/f4a0485668f437182d08abdf4b33d5be1b7c42d1/pgmpy/models/SEM.py#L596 

I think the issue is only when we are converting a SEMGraph object to SEMAlg object because we represent the unspecified edge weight as nan in SEMGraph: https://github.com/pgmpy/pgmpy/blob/f4a0485668f437182d08abdf4b33d5be1b7c42d1/pgmpy/models/SEM.py#L102

File: pgmpy/models/SEM.py [hunk 2]
```diff
@@ -589,20 +589,20 @@ def to_lisrel(self):
...
-            self.err_graph, nodelist=nodelist, weight="weight"
+        err_fixed = np.nan_to_num(
+            nx.to_numpy_array(self.err_graph, nodelist=nodelist, weight="weight")
         )
 
         # Add the variance of the error terms.
         for index, node in enumerate(nodelist):
-            try:
-                err_fixed[index, index] = self.err_graph.nodes[node]["weight"]
-            except KeyError:
```
Comment (jihyeseo) [inline pgmpy/models/SEM.py:604]: We do not need KeyError, because the missing key was already set with nan in https://github.com/pgmpy/pgmpy/blob/f6897595ac657b273816d9d68aca82aff335348f/pgmpy/models/SEM.py#L127, for every node.

**Comment by ankurankan** [comment]
@jihyeseo Thanks for the PR. I think the test was failing because of random initialization, I have remove it for now.
